events/janus_gelfevh.c

```diff
@@ -0,0 +1,590 @@
+	 * here could dangerously slow Janus down. Let's just reference and enqueue the event,
+	 * and handle it in our own thread: the event contains a monotonic time indicator of
+	 * when the event actually happened on this machine, so that, if relevant, we can compute
+	 * any delay in the actual event processing ourselves. */
+	json_incref(event);
+	g_async_queue_push(events, event);
+
+}
+
+json_t *janus_gelfevh_handle_request(json_t *request) {
+	if(g_atomic_int_get(&stopping) || !g_atomic_int_get(&initialized)) {
+		return NULL;
+	}
+	/* We can use this requests to apply tweaks to the logic */
+	int error_code = 0;
+	char error_cause[512];
+	JANUS_VALIDATE_JSON_OBJECT(request, request_parameters,
+		error_code, error_cause, TRUE,
+		JANUS_GELFEVH_ERROR_MISSING_ELEMENT, JANUS_GELFEVH_ERROR_INVALID_ELEMENT);
+	if(error_code != 0)
+		goto plugin_response;
+	/* Get the request */
+	const char *request_text = json_string_value(json_object_get(request, "request"));
+	if(!strcasecmp(request_text, "tweak")) {
+		/* We only support a request to tweak the current settings */
+		JANUS_VALIDATE_JSON_OBJECT(request, tweak_parameters,
+			error_code, error_cause, TRUE,
+			JANUS_GELFEVH_ERROR_MISSING_ELEMENT, JANUS_GELFEVH_ERROR_INVALID_ELEMENT);
+		if(error_code != 0)
+			goto plugin_response;
+		/* Parameters we can change */
+		const char *req_events = NULL, *req_backend = NULL, *req_port = NULL;
+		int req_compress = -1, req_compression = -1;
+		/* Events */
+		if(json_object_get(request, "events"))
+			req_events = json_string_value(json_object_get(request, "events"));
+		/* Compression */
+		if (json_object_get(request, "compress"))
+			req_compress = json_is_true(json_object_get(request, "compress"));
+		if (json_object_get(request, "compression"))
+			req_compress = json_integer_value(json_object_get(request, "compression"));
+		/* Backend stuff */
+		if(json_object_get(request, "backend"))
+			req_backend = json_string_value(json_object_get(request, "backend"));
+		if(json_object_get(request, "port"))
+			req_port = json_string_value(json_object_get(request, "port"));
+		if (json_object_get(request, "max_message_len"))
+			max_gelf_msg_len = json_integer_value(json_object_get(request, "max_message_len"));
+		if (strcasecmp(json_string_value(json_object_get(request, "protocol")), "udp") == 0){
+			transport = JANUS_GELFEVH_SOCKET_TYPE_UDP;
+		} else {
+			transport = JANUS_GELFEVH_SOCKET_TYPE_TCP;
+		}
+		if(!req_backend || !req_port) {
+			/* Invalid backend address or port */
+			error_code = JANUS_GELFEVH_ERROR_INVALID_ELEMENT;
+			g_snprintf(error_cause, sizeof(error_cause), "Invalid backend URI '%s:%s'", req_backend, req_port);
+			goto plugin_response;
+		}
+		/* If we got here, we can enforce */
+		janus_mutex_lock(&evh_mutex);
+		if(req_events)
+			janus_events_edit_events_mask(req_events, &janus_gelfevh.events_mask);
+		if (req_compress > -1)
+			compress = req_compress ? TRUE : FALSE;
```

Comment:
Same considerations about the warning here on enabling compression when TCP is used.